### PR TITLE
chapter_10

### PR DESCRIPTION
**Pull request by azimfaiz**

### CODE REVIEW

**Comment by azimfaiz** [comment]
ready

File: Gemfile [hunk 1]
```diff
@@ -14,6 +14,9 @@ gem "bootsnap", ">= 1.4.2", require: false
 gem "config"
 gem "bootstrap-sass" 
 gem "bcrypt", "3.1.13"
+gem "faker" , "2.1.2"
+gem "will_paginate", "3.1.8"
+gem "bootstrap-will_paginate", "1.0.0"
```
Comment (saobv-2223) [inline Gemfile]: using gem kaminari instead of gem will_paginate

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -14,14 +23,67 @@ def create
...
+      redirect_to login_url
+    end
+  end
+
+  def load_user
+    @user = User.find_by params[:id]
+    return unless @user
+    
+      flash[:danger] = "User not found"
+      redirect_to root_url
```
Comment (saobv-2223) [inline app/controllers/users_controller.rb]:
```suggestion
    return unless @user
    
    flash[:danger] = "User not found"
    redirect_to root_url
```

File: app/views/users/edit.html.erb [hunk 3]
```diff
@@ -0,0 +1,28 @@
+
+<% provide :title, "Edit user" %>
+<h1>Update your profile</h1>
```
Comment (saobv-2223) [inline app/views/users/edit.html.erb]:
```suggestion
<% provide :title, "Edit user" %>
<h1>Update your profile</h1>
```

File: app/views/users/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,11 @@
+<% provide(:title, 'All users') %>
+<h1>All users</h1>
+
+<%= will_paginate %>
```
Comment (saobv-2223) [inline app/views/users/index.html.erb]: using gem kaminari to paginate

**Comment by azimfaiz** [comment]
ready

File: app/assets/stylesheets/custom.scss [hunk 5]
```diff
@@ -165,7 +165,7 @@ input {
 
 #error_explanation {
   color:$red;
-
+
```
Comment (chungpv-1008) [inline app/assets/stylesheets/custom.scss]:
```suggestion

```

File: app/controllers/users_controller.rb [hunk 6]
```diff
@@ -14,14 +23,67 @@ def create
       flash[:success] = "Welcome to the Sample App!"
       redirect_to @user
     else
-      flash.now[:danger] = "Please try again!"
+      flash[:danger] = "Please try again!"
```
Comment (chungpv-1008) [inline app/controllers/users_controller.rb]:
```suggestion
      flash.now[:danger] = "Please try again!"
```

File: app/controllers/users_controller.rb [hunk 7]
```diff
@@ -14,14 +23,67 @@ def create
...
   end
 
+  def edit
+    @user = User.find_by params[:id]
+    if @user.edit user_params
+      flash.now[:success] = "User found"
+      render :edit
+    else
+      flash[:danger] = "User not found"
+      redirect_to root_url
```
Comment (chungpv-1008) [inline app/controllers/users_controller.rb]:
```suggestion
    return if @user
    
    flash[:danger] = "User not found"
    redirect_to root_url
```

File: app/controllers/users_controller.rb [hunk 8]
```diff
@@ -14,14 +23,67 @@ def create
...
+    end  
+  end
+
+  def update
+    @user = User.find_by params[:id]
+    if @user.update user_params
+      flash.now[:success] = "Profile updated"
+      redirect_to @user
+    else
+      flash[:success] = "Profile not updated"
```
Comment (chungpv-1008) [inline app/controllers/users_controller.rb]:
```suggestion
      flash[:success] = "Profile updated"
      redirect_to @user
    else
      flash.now[:success] = "Profile not updated"
```

**Comment by azimfaiz** [comment]
ready

**Review by saobv-2223 — COMMENTED** [review]
merge

**Pull request merged**